[MRG] DOC cleaning up what's new for 0.19

jnothman opened the pull request:
subsections, reclassifications, etc

Marked WIP only because of optimistic inclusion of multiple metrics, which is not yet merged.

jnothman committed:
DOC cleaning up what's new for 0.19

jnothman committed:
More cleaning up

jnothman committed:
More cleaning up

jnothman committed:
Deprecations

jnothman committed:
Merge branch 'master' into whatsnew-0.19

jnothman committed:
Clean up merge

jnothman committed:
Merge branch 'master' into whatsnew-0.19

jnothman committed:
Update

ogrisel commented on doc/whats_new.rst:
```diff
@@ -157,342 +149,442 @@ Enhancements
-     :issue:`9105` by `Hanmin Qin <https://github.com/qinhanmin2014>`_.
+   - :class:`neighbors.LSHForest` has been deprecated and will be
+     removed in 0.21 due to poor performance.
+     :issue:`8996` by `Andreas Müller`_.
```
This item does not seem to belong to "Linear, kernelized and related models".

amueller replied:
issue by me, but not the fix, so probably should be a different name?

ogrisel commented on doc/whats_new.rst:
```diff
@@ -157,342 +149,442 @@ Enhancements
-     would fail with a unmeaningful error by default.
-     :issue:`6573` by :user:`Quazi Marufur Rahman <qmaruf>` and
-     `Manoj Kumar`_.
+Linear, kernelized and related models
```
Maybe we should have one section name "Neighborhood models" and another "Semi-supervised learning models" instead?

jnothman commented:
I am fine to make more or different headings if it is helpful.

On 5 Jul 2017 12:15 am, "Olivier Grisel" <notifications@github.com> wrote:

> *@ogrisel* commented on this pull request.
> ------------------------------
>
> In doc/whats_new.rst
> <https://github.com/scikit-learn/scikit-learn/pull/9252#discussion_r125482929>
> :
>
> >
> -  -  Fixed a bug in :class:`gaussian_process.GaussianProcessRegressor`
> -     when the standard deviation and covariance predicted without fit
> -     would fail with a unmeaningful error by default.
> -     :issue:`6573` by :user:`Quazi Marufur Rahman <qmaruf>` and
> -     `Manoj Kumar`_.
> +Linear, kernelized and related models
>
> Maybe we should have one section name "Neighborhood models" and another
> "Semi-supervised learning models" instead?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/scikit-learn/scikit-learn/pull/9252#pullrequestreview-47876144>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAEz6xLGkDt9e2iSoj3KyrSfjZRW0y0Jks5sKkkGgaJpZM4OKFjl>
> .
>

jnothman commented:
of course could just group by subpackage

On 5 Jul 2017 7:39 am, "Joel Nothman" <joel.nothman@gmail.com> wrote:

> I am fine to make more or different headings if it is helpful.
>
> On 5 Jul 2017 12:15 am, "Olivier Grisel" <notifications@github.com> wrote:
>
>> *@ogrisel* commented on this pull request.
>> ------------------------------
>>
>> In doc/whats_new.rst
>> <https://github.com/scikit-learn/scikit-learn/pull/9252#discussion_r125482929>
>> :
>>
>> >
>> -  -  Fixed a bug in :class:`gaussian_process.GaussianProcessRegressor`
>> -     when the standard deviation and covariance predicted without fit
>> -     would fail with a unmeaningful error by default.
>> -     :issue:`6573` by :user:`Quazi Marufur Rahman <qmaruf>` and
>> -     `Manoj Kumar`_.
>> +Linear, kernelized and related models
>>
>> Maybe we should have one section name "Neighborhood models" and another
>> "Semi-supervised learning models" instead?
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/scikit-learn/scikit-learn/pull/9252#pullrequestreview-47876144>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe-auth/AAEz6xLGkDt9e2iSoj3KyrSfjZRW0y0Jks5sKkkGgaJpZM4OKFjl>
>> .
>>
>

jnothman commented:
Unless we want to completely group by subpackage, which might make sense but seems unnecessarily  fine-grained for this purpose, I've just added "other predictors".

I had included neighbors and semi_supervised under "linear, kernelised et al" because of how they use affinities. But so does clustering. And so do kernels. And most of our decompositions are linear. So we're splitting hairs in any case. Would you rather I group by subpackage?

jnothman committed:
TODOs to prose and minor changes

jnothman committed:
Merge branch 'master' into whatsnew-0.19

jnothman committed:
Changed models and minor fixes

jnothman committed:
sort

amueller commented:
can you merge it with the 0.18.2 whatsnew?

jnothman committed:
Merge in 0.18.2 docs

jnothman committed:
Missing entry from 0.18 logs

jnothman committed:
Optimistically add some features to highlights

lesteve commented on doc/whats_new.rst:
```diff
@@ -505,56 +617,46 @@ API changes summary
-   - The ``n_topics`` parameter of :class:`decomposition.LatentDirichletAllocation`
-     has been renamed to ``n_components`` and will be removed in version 0.21.
-     :issue:`8922` by :user:`Attractadore`
+Metrics
```
This section is empty, should it be removed?

lesteve committed:
Forgotten user directive

lesteve committed:
Fix alignment

lesteve commented:
I push a couple of changes but apart from the empty "Metrics" section, LGTM.

amueller commented on doc/modules/classes.rst:
```diff
@@ -1391,6 +1389,18 @@ Recently deprecated
 ===================
 
 
+To be removed in 0.21
```
nitpickiest nitpick: shouldn't that go below removed in 0.20? though no strong opinion.

jnothman replied:
Mostly the point of listing them is so that autodoc generates. Order shmorder.

amueller commented on doc/whats_new.rst:
```diff
@@ -10,6 +10,39 @@ Version 0.19
+We are excited to release a number of great new features including
+:class:`neighbors.LocalOutlierFactor` for anomaly detection,
+:class:`preprocessing.QuantileTransformer` for robust feature
+transformation, and :class:`multioutput.ClassifierChain` to simply
```
Maybe say that ClassifierChain is a meta-estimator?

amueller commented on doc/whats_new.rst:
```diff
@@ -10,6 +10,39 @@ Version 0.19
+account for dependencies between classes in multilabel problems. We
+have some new algorithms in existing estimators, such as
+multiplicative update in :class:`decomposition.NMF` and multinomial
+:class:`linear_model.LogisticRegression` with L1 loss.
```
via the saga solver?

amueller commented on doc/whats_new.rst:
```diff
@@ -10,6 +10,39 @@ Version 0.19
+you can predict faster: if you're sure you know what you're doing, you can turn
+off validating that the input is finite using :func:`config_context`.
+
+Cross validation is now able to return the results from multiple metric
```
that's not merged yet, right?

jnothman replied:
no, it's not. Nor are the TSNE fixes. Being optimisitc, and getting the writing out of the way.

amueller commented on doc/whats_new.rst:
```diff
@@ -10,6 +10,39 @@ Version 0.19
+:class:`manifold.TSNE` implementation have been fixed, particularly in the
+default Barnes-Hut approximation.  :class:`semi_supervised.LabelSpreading` and
+:class:`semi_supervised.LabelPropagation` have had substantial fixes.
+Propagation was previously broken. Spreading should now correctly respect its
```
LabelPropagation?

amueller commented on doc/whats_new.rst:
```diff
@@ -65,434 +97,514 @@ New features
+   - Fixed a bug in :class:`linear_model.RandomizedLasso`,
+     :class:`linear_model.Lars`, :class:`linear_model.LassoLars`,
+     :class:`linear_model.LarsCV` and :class:`linear_model.LassoLarsCV`,
+     where the parameter ``precompute`` were not used consistently across
```
wast not?

amueller commented on doc/whats_new.rst:
```diff
@@ -65,434 +97,514 @@ New features
-     where a float being compared to ``0.0`` using ``==`` caused a divide by zero
-     error. issue:`7970` by :user:`He Chen <chenhe95>`.
+   - Fix a bug where :func:`linear_model.LassoLars.fit` sometimes
+     left `coef_` as a list, rather than an ndarray.
```
double backticks?

amueller commented on doc/whats_new.rst:
```diff
@@ -65,434 +97,514 @@ New features
-     array X and initial centroids, where X's means were unnecessarily being
-     subtracted from the centroids. :issue:`7872` by :user:`Josh Karnofsky <jkarno>`.
+   - Fix :func:`linear_model.BayesianRidge.fit` to return
+     ridge parameter `alpha_` and `lambda_` consistent with calculated
```
double backtricks everywhere?

amueller commented on doc/whats_new.rst:
```diff
@@ -65,434 +97,514 @@ New features
      This also impacts the output shape of :class:`decomposition.DictionaryLearning`.
      :issue:`8086` by `Andreas Müller`_.
 
+   - Fixed the implementation of `explained_variance_`
```
double backticks?

amueller reviewed (commented):
Are you sure we should release? That really seems like to little to warrant a new version ;)

jnothman committed:
Cleaning up for Andy's comments

jnothman committed:
Merge branch 'master' into whatsnew-0.19

jnothman committed:
Mention beta_loss=0 speedup

jnothman commented:
> Are you sure we should release? That really seems like to little to warrant a new version ;)

Maybe we've missed something? Make sure you go back and check the logs before giving this LGTM. :P

jnothman committed:
Merge branch 'whatsnew-0.19' of github.com:jnothman/scikit-learn into whatsnew-0.19

jnothman committed:
Merge branch 'master' into whatsnew-0.19

jnothman committed:
Update

amueller commented:
why are there conflicts in the tsne test?

jnothman commented:
Weird. Must have been a merge error on my part. I'll fix it up shortly

jnothman committed:
Merge branch 'master' into whatsnew-0.19

jnothman committed:
Clean up new what's new entries

jnothman commented:
Marking MRG

jnothman committed:
DOC Add changes missed from what's new

And other minor things.

This took lots of effort which I would have not committed where I not home sick...

jnothman commented:
Okay. Let's do a release, then :)
